recipe/build-pkg.sh

```diff
@@ -16,6 +16,11 @@ else
     FAISS_ENABLE_GPU="OFF"
 fi
 
+if [ ${python_impl} == "pypy" ]; then
+    # needed by patched CMakeLists.txt for correct PyPy-extension
+    export EXTRA_SO_EXTENSION=".pypy36-pp73-x86_64-linux-gnu"
+fi
```

Comment:
Is there a way to make this more general? Ahh, now I remember some issue with CMake with this, perhaps https://gitlab.kitware.com/cmake/cmake/-/issues/21070 ?

Reply:
Would love for this to be more generic of course (aside from the fact that my patch didn't work yet... 😒). Not sure if I'm following your comments in the cmake-issue - `EXT_SUFFIX` is not set in the CI job (AFAICT), so how can I reuse it from `build-pkg.sh`, resp. within `CMakeLists.txt`?

Reply:
I don't really know CMake. I think you need to use [`Python_SOABI`](https://cmake.org/cmake/help/latest/module/FindPython.html) (about 1/4 way down that page) somewhere to tell [SWIG's CMake file](https://github.com/facebookresearch/faiss/blob/master/faiss/python/CMakeLists.txt#L45) about the output module name. I would expect that to happen automatically by SWIG, but [this part](http://www.swig.org/Doc4.0/Python.html#Python_nn7) of the SWIG documentation seems to indicate otherwise.

I couldn't find `EXTRA_SO_EXTENSION` in the CMake documentation, how does that end up getting used?

In any case, I think this could become 
`export EXTRA_SO_EXTENSION=$(python -c "import sysconfig; print(sysconfig.get_config_var('EXT_SUFFIX'))")`
It would not suffer from the pre-cpython3.8 issue I linked to on Windows since this script is only used for non-windows, so you shouldn't need the conditional `python_impl`. Probably worthy of a comment though, something like `# Override SWIG c-extension shared-object name _swigfaiss.so with something like _swigfaiss..cpython-38-x86_64-linux-gnu.so`